File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(auto command : commands) {
        int s = command[0] - 1;
        int e = command[1];
        int k = command[2] - 1;
        
        vector<int> tmp(array.begin() + s, array.begin() + e);  // 부분 배열 복사
        
        sort(tmp.begin(), tmp.end());
        
        answer.push_back(tmp[k]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer> list = new ArrayList<>();
        
        for(int[] command : commands) {
            int i = command[0];
            int j = command[1];
            int k = command[2];
            
            // 배열 자르기: Arrays.copyOfRange(배열, 시작 인덱스, 종료 인덱스)
            int[] arr = Arrays.copyOfRange(array, i - 1, j);
            Arrays.sort(arr);
            
            list.add(arr[k - 1]);
        }
        
        return list.stream().mapToInt(Integer::intValue).toArray();
    }
}